File: android/app/src/main/java/com/desmoines/aipulse/data/model/ContentHub.kt
package com.desmoines.aipulse.data.model

import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Forest
import androidx.compose.material.icons.filled.MusicNote
import androidx.compose.material.icons.filled.SportsBasketball
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector

/**
 * A curated content hub (ANDP-027). Music / Sports / Outdoors each map to a
 * config that drives one generic ContentHubScreen — mixing relevant events,
 * themed attractions, and featured dining. Mirrors iOS ContentHub.
 */
enum class ContentHub(
    val id: String,
    val title: String,
    val blurb: String,
    val icon: ImageVector,
    val gradient: List<Color>,
    /** Event category/title terms (OR-matched, case-insensitive). */
    val eventTerms: List<String>,
    val eventsSectionTitle: String,
    /** Attraction types for the "places" rail (empty → no rail). */
    val attractionTypes: List<AttractionType>,
    val attractionsSectionTitle: String,
    val diningSectionTitle: String,
) {
    MUSIC(
        id = "music",
        title = "Live Music",
        blurb = "Your guide to the Des Moines music scene — tonight's shows, venue guides, and upcoming concerts all in one place.",
        icon = Icons.Filled.MusicNote,
        gradient = listOf(Color(0xFF8C5CF5), Color(0xFFA854F7)),
        eventTerms = listOf("Music", "Concert", "Live Music"),
        eventsSectionTitle = "Upcoming shows",
        attractionTypes = listOf(AttractionType.ENTERTAINMENT),
        attractionsSectionTitle = "Venues & entertainment",
        diningSectionTitle = "Dinner before the show",
    ),
    SPORTS(
        id = "sports",
        title = "Sports",
        blurb = "Des Moines is one of the best minor-league sports markets in America. From Iowa Cubs baseball to Iowa Wild hockey, find your next gameday.",
        icon = Icons.Filled.SportsBasketball,
        gradient = listOf(Color(0xFF10B883), Color(0xFF159178)),
        eventTerms = listOf("Sport", "Game", "Baseball", "Hockey", "Basketball", "Football", "Soccer"),
        eventsSectionTitle = "Upcoming games",
        attractionTypes = listOf(AttractionType.SPORTS),
        attractionsSectionTitle = "Venues & arenas",
        diningSectionTitle = "Gameday eats",
    ),
    OUTDOORS(
        id = "outdoors",
        title = "Trails & Outdoors",
        blurb = "Explore the metro's incredible parks and 800+ miles of trails — from the High Trestle Trail Bridge to peaceful urban loops.",
        icon = Icons.Filled.Forest,
        gradient = listOf(Color(0xFF21C55E), Color(0xFF17A34A)),
        eventTerms = listOf("Outdoor", "Festival", "Nature", "Park", "Hike", "Trail", "Market"),
        eventsSectionTitle = "Outdoor events",
        attractionTypes = listOf(AttractionType.PARK, AttractionType.GARDEN, AttractionType.ZOO),
        attractionsSectionTitle = "Parks & nature",
        diningSectionTitle = "Patios & local bites",
    );

    /** Whether [event] belongs to this hub by category or title term match. */
    fun matchesEvent(event: Event): Boolean = eventTerms.any { term ->
        event.category?.contains(term, ignoreCase = true) == true ||
            event.title.contains(term, ignoreCase = true)
    }

    /** Whether [attraction] belongs to this hub's places rail. */
    fun matchesAttraction(attraction: Attraction): Boolean =
        attractionTypes.contains(attraction.attractionType)

    companion object {
        fun byId(id: String): ContentHub? = entries.firstOrNull { it.id == id }
    }
}

File: ios/DesMoinesInsider/Models/ContentHub.swift
import SwiftUI

/// A curated content hub (IOS-PARITY-006). Music / Sports / Outdoors each map to
/// a `ContentHub` config that drives one generic `ContentHubView` — mixing
/// upcoming relevant events, themed attractions, and featured dining, parity
/// with the web /music, /sports, /outdoors hubs.
enum ContentHub: String, CaseIterable, Identifiable {
    case music
    case sports
    case outdoors

    var id: String { rawValue }

    /// Maps from the Discover-hub destination so deep links route here.
    init?(destination: DiscoverDestination) {
        switch destination {
        case .music: self = .music
        case .sports: self = .sports
        case .outdoors: self = .outdoors
        default: return nil
        }
    }

    var title: String {
        switch self {
        case .music: return "Live Music"
        case .sports: return "Sports"
        case .outdoors: return "Trails & Outdoors"
        }
    }

    /// Editorial blurb shown in the hero (parity with the web hub intros).
    var blurb: String {
        switch self {
        case .music:
            return "Your guide to the Des Moines music scene — tonight's shows, venue guides, and upcoming concerts all in one place."
        case .sports:
            return "Des Moines is one of the best minor-league sports markets in America. From Iowa Cubs baseball to Iowa Wild hockey, find your next gameday."
        case .outdoors:
            return "Explore the metro's incredible parks and 800+ miles of trails — from the High Trestle Trail Bridge to peaceful urban loops."
        }
    }

    var systemImage: String {
        switch self {
        case .music: return "music.note"
        case .sports: return "sportscourt.fill"
        case .outdoors: return "leaf.fill"
        }
    }

    var gradient: [Color] {
        switch self {
        case .music: return [Color(red: 0.55, green: 0.36, blue: 0.96), Color(red: 0.66, green: 0.33, blue: 0.97)]
        case .sports: return [Color(red: 0.06, green: 0.72, blue: 0.51), Color(red: 0.08, green: 0.57, blue: 0.47)]
        case .outdoors: return [Color(red: 0.13, green: 0.77, blue: 0.37), Color(red: 0.09, green: 0.64, blue: 0.29)]
        }
    }

    /// Event `category` ilike terms (OR-matched), mirroring the web hub queries.
    var eventTerms: [String] {
        switch self {
        case .music: return ["Music", "Concert", "Live Music"]
        case .sports: return ["Sport", "Game", "Baseball", "Hockey", "Basketball", "Football", "Soccer"]
        case .outdoors: return ["Outdoor", "Festival", "Nature", "Park", "Hike", "Trail", "Market"]
        }
    }

    var eventsSectionTitle: String {
        switch self {
        case .music: return "Upcoming shows"
        case .sports: return "Upcoming games"
        case .outdoors: return "Outdoor events"
        }
    }

    /// Attraction `type` raw values for this hub's "places" rail (empty → no rail).
    var attractionTypes: [AttractionType] {
        switch self {
        case .music: return [.entertainment]
        case .sports: return [.sports]
        case .outdoors: return [.park, .garden, .zoo]
        }
    }

    var attractionsSectionTitle: String {
        switch self {
        case .music: return "Venues & entertainment"
        case .sports: return "Venues & arenas"
        case .outdoors: return "Parks & nature"
        }
    }

    /// Featured-dining rail title (all hubs include a dining cross-sell).
    var diningSectionTitle: String {
        switch self {
        case .music: return "Dinner before the show"
        case .sports: return "Gameday eats"
        case .outdoors: return "Patios & local bites"
        }
    }
}
